overviewer_core/textures.py

```diff
@@ -1994,12 +2020,18 @@ def composter(self, blockid, data):
     alpha_over(img, img2, (0, 0), img2)
     return img
 
-# fire
-@material(blockid=51, data=list(range(16)), transparent=True)
+# fire and soul_fire
+@material(blockid=[51, 1040], data=list(range(16)), transparent=True)
 def fire(self, blockid, data):
-    firetextures = self.load_fire()
-    side1 = self.transform_image_side(firetextures[0])
-    side2 = self.transform_image_side(firetextures[1]).transpose(Image.FLIP_LEFT_RIGHT)
+    if blockid == 51:
+        firetextures = self.load_fire()
+        side1 = self.transform_image_side(firetextures[0])
+        side2 = self.transform_image_side(firetextures[1]).transpose(Image.FLIP_LEFT_RIGHT)
+    elif blockid == 1040:
+        soul_firetextures = self.load_soul_fire()
+        side1 = self.transform_image_side(soul_firetextures[0])
+        side2 = self.transform_image_side(soul_firetextures[1]).transpose(Image.FLIP_LEFT_RIGHT)
+
```

Comment:
I don't think you need to duplicate the transform code here.

```python
firetextures = self.load_fire() if blockid == 51 else self.load_soul_fire()
side1 = self.transform_image_side(firetextures[0])
side2 = self.transform_image_side(firetextures[1]).transpose(Image.FLIP_LEFT_RIGHT)
```
should be enough to handle both cases.

Reply:
Hi @CounterPillow 

Unless i completely missed something about the ``data`` parameter, this is always zero for both fire variants ( not 0 .. 15)
So this can be simplified to

```python
# fire and soul_fire
@material(blockid=[51, 1040], transparent=True)
def fire(self, blockid, data):
    if blockid == 51:
        textureNS = self.load_image_texture("assets/minecraft/textures/block/fire_0.png")
        textureEW = self.load_image_texture("assets/minecraft/textures/block/fire_1.png")
    elif blockid == 1040:
        textureNS = self.load_image_texture("assets/minecraft/textures/block/soul_fire_0.png")
        textureEW = self.load_image_texture("assets/minecraft/textures/block/soul_fire_1.png")
    side1 = self.transform_image_side(textureNS)
    side2 = self.transform_image_side(textureEW).transpose(Image.FLIP_LEFT_RIGHT)
    img = Image.new("RGBA", (24,24), self.bgcolor)
    alpha_over(img, side1, (12,0), side1)
    alpha_over(img, side2, (0,0), side2)
    alpha_over(img, side1, (0,6), side1)
    alpha_over(img, side2, (12,6), side2)    
    return img
``` 

Or am I missing something?